src/Automation/ScanResultsAssembler.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using Axe.Windows.Core.Bases;
+using Axe.Windows.Core.Results;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Axe.Windows.Automation
+{
+    /// <summary>
+    /// Provides methods used to assemble a <see cref="ScanResults"/> object
+    /// </summary>
+    public static class ScanResultsAssembler
+    {
+        /// <summary>
+        /// Assembles failed scans from the provided element
+        /// </summary>
+        /// <param name="element">Root element from which scan results will be assembled</param>
+        /// <returns>A ScanResults object containing the relevant errors and error count</returns>
+        public static ScanResults AssembleScanResultsFromElement(A11yElement element)
```

Comment:
>A11yElement [](start = 65, length = 11)

Should this be ICoreA11yElement instead of A11yElement?

Reply:
`ICoreA11yElement` is internal to Axe.Windows.Core.Bases. And doesn't expose Properties, Patterns, or ScanResults. Is there something I'm missing about how I could use it here?